src/app/home/home.component.ts

```diff
@@ -134,23 +134,31 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
     const user_id = 'org.couchdb.user:' + this.userService.get().name;
     this.couchService.allDocs('notifications')
       .subscribe((data) => {
-        let cnt = 0;
         data.sort((a, b) => 0 - (new Date(a.time) > new Date(b.time) ? 1 : -1));
         this.notifications = data.map(notifications => {
-          if (notifications.status === 'unread') {
-            cnt ++;
-          }
-          return notifications;
+         return Object.keys(notifications).map(key => {
+            return { [ key ] : notifications[ key ] };
+          });
         }).filter(nt  => {
-          return nt.user === user_id;
+          return nt[ 2 ].user === user_id;

```

Comment:
now you have started to add number as key. I think the problem is not clear with you.